static/app/utils/replays/replayDataUtils.tsx

```diff
@@ -41,16 +41,18 @@ export function mapResponseToReplayRecord(apiResponse: any): ReplayRecord {
   // Add missing tags to the response
   const unorderedTags: ReplayRecord['tags'] = {
     ...apiResponse.tags,
-    ...(apiResponse.os?.name ? {'os.name': [apiResponse.os.name]} : {}),
-    ...(apiResponse.os?.version ? {'os.version': [apiResponse.os.version]} : {}),
     ...(apiResponse.browser?.name ? {'browser.name': [apiResponse.browser.name]} : {}),
     ...(apiResponse.browser?.version
       ? {'browser.version': [apiResponse.browser.version]}
       : {}),
-    ...(apiResponse.device?.name ? {'device.name': [apiResponse.device.name]} : {}),
-    ...(apiResponse.device?.family ? {'device.family': [apiResponse.device.family]} : {}),
     ...(apiResponse.device?.brand ? {'device.brand': [apiResponse.device.brand]} : {}),
+    ...(apiResponse.device?.family ? {'device.family': [apiResponse.device.family]} : {}),
```

Comment:
Hmm, I wonder if our backend should be flattening these keys for us. Our other APIs I think do e.g. the response will have a key of `device.family`.

Reply:
This code is flattening the key, but also sticking the value into the `tags` object. 

So we start with something like:
```
const apiResponse = {
  id: "56832378a54f49dba046f909f80cb17f",
  startedAt: "2022-10-31T11:44:12+00:00"
  device: {
    family: "Mac"
  },
  tags: {}
};
```

and end with:
```
return {
  id: "56832378a54f49dba046f909f80cb17f",
  startedAt: "2022-10-31T11:44:12+00:00"
  device: {
    family: "Mac"
  },
  tags: {
    "device.family": "Mac"
  }
};
```

that makes it easier to populate the tags table later on